### PR TITLE
Allow Faraday 1.x

### PR DESCRIPTION
**Pull request by jfeltesse-mdsol**
The bulk of the "work" here was in the integration TestApp helper. It hanged for some reason (race?), and in any case the code was convoluted. After refactoring, no issues.

@adriancole @jcarres-mdsol @ykitamura-mdsol

### CODE REVIEW

**Comment by jfeltesse-mdsol** [comment]
Follow up question: should we push faraday 1.x or make it accept both 0.x or 1.x? I didn't go that route because we'd then need to test both versions using appraisal or wwtd.

**Comment by jcarres-mdsol** [comment]
I would imagine there is an universe of things out there depending on Faraday < 1 , would it take a while for libraries and applications to move to faraday 1.x?

**Comment by jfeltesse-mdsol** [comment]
@jcarres-mdsol added Appraisal to test both 0.x and 1.x. Should be good to go.

**Comment by ykitamura-mdsol** [comment]
looks good to me 👍

File: .travis.yml [hunk 1]
```diff
@@ -18,12 +25,12 @@ deploy:
   on:
     tags: true
     repo: openzipkin/zipkin-ruby
-    condition: "$TRAVIS_RUBY_VERSION == 2.6"
+    condition: "$TRAVIS_RUBY_VERSION == 2.7"
```
Comment (ykitamura-mdsol) [inline .travis.yml:28]: Travis is trying to deploy on both gemfiles...
https://travis-ci.org/github/openzipkin/zipkin-ruby/jobs/675596484

**Pull request merged**